appcues/src/main/java/com/appcues/statemachine/states/BeginningExperienceState.kt

```diff
@@ -0,0 +1,29 @@
+package com.appcues.statemachine.states
+
+import com.appcues.data.model.Experience
+import com.appcues.statemachine.Action
+import com.appcues.statemachine.Action.StartExperience
+import com.appcues.statemachine.State
+import com.appcues.statemachine.Transition
+import com.appcues.statemachine.effects.PresentationEffect
+
+internal data class BeginningExperienceState(val experience: Experience) : State {
+
+    override val currentExperience: Experience
+        get() = experience
+    override val currentStepIndex: Int?
+        get() = null
+
+    override fun take(action: Action): Transition? {
+        return if (action is StartExperience) {
```

Comment:
beginning experience takes StartExperience (same from IdlingState) since it will always start at Step 0, we can then use StartStep for a more specific use case later.